frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx

```diff
@@ -193,24 +222,60 @@ function DashCardVisualization({
           isResultDirty: false,
         }));
 
+    const handleModalOpen = async () => {
+      try {
+        if (!chartPublicUuid) {
+          const public_uuid = await getChartUuid(dashcard.card.id);
+          setChartPublicUuid(public_uuid);
+        }
+        setModalOpen(true);
+      } catch (error) {
+        console.error("Error fetching chart API data:", error);
+      }
+    };
+
+    const handleModalClose = () => {
+      setModalOpen(false);
+    };
+
     if (!shouldShowDownloadWidget) {
       return null;
     }
 
     return (
-      <CardDownloadWidget
-        className={SAVING_CHART_IMAGE_HIDDEN_CLASS}
-        classNameClose="hover-child hover-child--smooth"
-        card={dashcard.card}
-        result={mainSeries}
-        params={parameterValuesBySlug}
-        dashcardId={dashcard.id}
-        dashboardId={dashboard.id}
-        token={isEmbed ? dashcard.dashboard_id : undefined}
-        icon="ellipsis"
-        iconSize={17}
-        hideWaterMark={hideWaterMark}
-      />
+      <div className="flex align-center">
+        <CardApiWidget
+          small
+          color={color("filter")}
+          onClick={handleModalOpen}
+          active={isModalOpen}
+          isNightMode={isNightMode}
+        >
+          {t`API`}
+        </CardApiWidget>
+
+        {isModalOpen && (
+          <DashCardApiModal
+            isNightMode={isNightMode}
+            isModalOpen={isModalOpen}
+            handleModalClose={handleModalClose}
+            chartPublicUuid={chartPublicUuid}
+          />
+        )}
+        <CardDownloadWidget
```

Comment:
There error throwing while using `CardDownloadWidget`
as `Property 'bordered' is missing in type`